### PR TITLE
[CN-499] Add Documentation for Restore from HotBackup CR

### PR DESCRIPTION
**Pull request by mtyazici**

### CODE REVIEW

File: docs/modules/ROOT/pages/backup-restore.adoc [hunk 1]
```diff
@@ -364,26 +378,51 @@ be deployed with the Hazelcast container in the same Pod. The agent will start a
...
++
+--
+[source,yaml,subs="attributes+"]
+----
+include::ROOT:example$/hazelcast-persistence-restore-cr-name.yaml[]
+----
+
+<1> The `HotBackup` resource name that will be used for restore. The bucket URI and secret will be taken from the `HotBackup` resource.
+<2> The agent which is responsible for restoring data from the external storage.
+    The agent configuration is optional. If you give `restore` under the `persistence` and do not pass the agent configuration, the operator
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/backup-restore.adoc:414]:
```suggestion
    The agent configuration is optional. If you provide `restore` under `persistence` and do not pass the agent configuration, Hazelcast Platform Operator
```

File: docs/modules/ROOT/pages/backup-restore.adoc [hunk 2]
```diff
@@ -364,26 +378,51 @@ be deployed with the Hazelcast container in the same Pod. The agent will start a
...
+    will use the latest agent version compatible with the operator version.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/backup-restore.adoc:415]:
```suggestion
    uses the latest agent version that is compatible with its version.
```

File: docs/modules/ROOT/pages/backup-restore.adoc [hunk 3]
```diff
@@ -364,26 +378,51 @@ be deployed with the Hazelcast container in the same Pod. The agent will start a
...
+
 . Apply the `Hazelcast` resource:
 +
 [source,bash]
 ----
-kubectl apply -f ./hazelcast-persistence-restore.yaml
+kubectl apply -f ./hazelcast-persistence-restore-external.yaml
 ----
 
 . See that the same PVCs are bound to the new pods:
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/backup-restore.adoc:428]:
```suggestion
. You can check that the same PVCs are bound to the new pods:
```

**Review by angelasimms — CHANGES_REQUESTED** [review]
Mainly minor edits but just a couple of points around consistency:

- I've checked and we use Hazelcast Platform Operator rather than operator etc.
- Where you have an instruction next to a code snippet, if the instruction is a full sentence then you don't need a colon. 

Examples:

Apply a resource.

Restore a resource.

The only time you need to use a colon is for a incomplete sentence.

For example:

Use the following command:

File: docs/modules/ROOT/pages/backup-restore.adoc [hunk 4]
```diff
@@ -132,7 +132,12 @@ hot-backup   Success
 
 === Restoring from Local Backups
 
-To restore a cluster from local backups, you can directly reapply the `Hazelcast` resource, and it will have access to the PVCs that contain the persisted data.
+To restore a cluster from local backups, you can directly reapply the `Hazelcast` resource, and it will have access to the PVCs that contain the persisted data. This will restore the Hazelcast cluster from existing `hot-restart` folders.
+
+However, if you want to restore from local backups you have taken using the `HotBackup` resource, you can give the `HotBackup` resource name in the restore configuration. For restore to work correctly, make sure the status of the `HotBackup` resource is `Success`. 
+
+When restore mechanism is used, the Restore Agent container will
+be deployed with the Hazelcast container in the same Pod. The agent will start as an initContainer before the Hazelcast container.
```
[the remark this answers is not in the document]
Comment (mtyazici) [reply docs/modules/ROOT/pages/backup-restore.adoc:140]: Should we remove `be` here?

File: docs/modules/ROOT/pages/backup-restore.adoc [hunk 5]
```diff
@@ -352,9 +366,9 @@ hot-backup   Success
 
 === Restoring from External Backups
 
-To restore a cluster from external backups, you must enable external restore by adding `restore configuration` under the persistence.
-Similar with External Persistence, you may pass the agent configuration. If not, default value of the agent configuration will be used.
-When external restore mechanism is used, the Restore Agent container will
+To restore a cluster from external backups, you can either set up the bucket configuration or give the `HotBackup` resource name you triggered the external backup. In either case backups will be restored from the external buckets.
+
+When restore mechanism is used, the Restore Agent container will
 be deployed with the Hazelcast container in the same Pod. The agent will start as an initContainer before the Hazelcast container.
```
[the remark this answers is not in the document]
Comment (mtyazici) [reply docs/modules/ROOT/pages/backup-restore.adoc:372]: Same as above

**Comment by angelasimms** [comment]
> Mainly minor edits but just a couple of points around consistency:
> 
> * I've checked and we use Hazelcast Platform Operator rather than operator etc.
> * Where you have an instruction next to a code snippet, if the instruction is a full sentence then you don't need a colon.
> 
> Examples:
> 
> Apply a resource.
> 
> Restore a resource.
> 
> The only time you need to use a colon is for a incomplete sentence.
> 
> For example:
> 
> Use the following command:

@mtyazici. I think you had a couple of queries on comments but they look like they have all been resolved. Do you need me to review again?

**Review by angelasimms — APPROVED** [review]
Looks fine now.

**Pull request merged**